### ISSUE TITLE
Restrict decoding

### ISSUE THREAD

**Issue by Kamil93** [kriszyp/cbor-x#14]
How can I restrict decoding a message in case where encoded message contain some values of types other than regular json supports (like ES6 Map/Set or sth), and discard it in decoded result, for example set it to `undefined`

**Comment by kriszyp** [comment]
Would it be preferable to skip these types of values on encoding instead of decoding (doesn't seem you would want to store them in the encoding if they will be skipped)? And would it be preferable to try to match JSON behavior where Maps and Sets are serialized as an empty object (Maps and Sets aren't treated as `undefined`s in JSON)?

**Comment by Kamil93** [comment]
Not always you have control in what is encoded, for example when it comes from client-side. Your proposal for empty object instead of undefined is more reasonable. Is it possible in cbor-x right now?

**Comment by kriszyp** [comment]
Yes. There isn't a single option to change everything, but you can override the handling of individual tags by using `addExtension` with the tags you want to change, and what you want them to return:
```
 // return {} for tag 258, used for Sets:
CBOR.addExtension({tag:258, decode(){return {}}})
// return {} for tag 1, used for Dates
CBOR.addExtension({tag:1, decode(){return {}}})
// return {} for tag 259, used for Maps
CBOR.addExtension({tag:259, decode(){return {}}})
```

**Comment by Kamil93** [comment]
Is there a list of tags that CBOR-X supports for js usage?

**Comment by kriszyp** [comment]
No, I don't have a list in one place, but I could add that to the docs.

**Comment by kriszyp** [comment]
Ok, I added the list of tags to the docs.
